File: CORE-FE/src/services/user.service.ts
import axiosInstance from '../axiosInstance';
import { type UserDTO, type UserSearchParams, type PageResponse } from '../types/user.types';

const USER_API_BASE = '/users';

export const userService = {
  // Create a new user
  createUser: async (userData: UserDTO): Promise<UserDTO> => {
    const response = await axiosInstance.post(USER_API_BASE, userData);
    return response.data.data;
  },

  // Update an existing user
  updateUser: async (id: number, userData: UserDTO): Promise<UserDTO> => {
    const response = await axiosInstance.put(`${USER_API_BASE}/${id}`, userData);
    return response.data.data;
  },

  // Delete a user
  deleteUser: async (id: number): Promise<void> => {
    await axiosInstance.delete(`${USER_API_BASE}/${id}`);
  },

  // Get user by ID
  getUserById: async (id: number): Promise<UserDTO> => {
    const response = await axiosInstance.get(`${USER_API_BASE}/${id}`);
    return response.data.data;
  },

  // Get all users by organization
  getAllUsers: async (organizationId: number): Promise<UserDTO[]> => {
    const response = await axiosInstance.get(USER_API_BASE, {
      params: { organizationId }
    });
    return response.data.data;
  },

  // Search users with pagination
  searchUsers: async (params: UserSearchParams): Promise<PageResponse<UserDTO>> => {
    const response = await axiosInstance.get(`${USER_API_BASE}/search`, {
      params: {
        organizationId: params.organizationId,
        keyword: params.keyword,
        page: params.page || 0,
        size: params.size || 10,
        sort: params.sort || 'username,asc'
      }
    });
    return response.data.data;
  }
};

File: core 2/src/main/java/com/dev/core/service/UserService.java
package com.dev.core.service;

import com.dev.core.model.UserDTO;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;
import java.util.Optional;

public interface UserService {

    UserDTO createUser(UserDTO dto);

    UserDTO updateUser(Long id, UserDTO dto);

    void deleteUser(Long id);

    UserDTO getUserById(Long id);

    Optional<UserDTO> getUserByEmail(String email);

    List<UserDTO> getAllUsers(Long organizationId);

    Page<UserDTO> searchUsers(Long organizationId, String keyword, Pageable pageable);
}
